libs/util/httpRequest.js

```diff
@@ -7,7 +7,7 @@
  * @module httpRequest
  */
 
-function FetchrError(message, options, request, response) {
+function FetchrError(reason, message, options, request, response) {
```

Comment:
probably one more argument away from just passing them all in as an object

Reply:
Yup, but it's a pain in the ass to work with objects as arguments in es5. I'll leave it for now and refactor when we move to ts.